ray-operator/controllers/ray/utils/validation_test.go

```diff
@@ -617,126 +617,154 @@ func TestValidateRayJobSpec(t *testing.T) {
+				Suspend:                  true,
+				ShutdownAfterJobFinishes: true,
+				RayClusterSpec:           createBasicRayClusterSpec(),
 			},
+			expectError: false,
 		},
-	})
-	require.ErrorContains(t, err, "the ClusterSelector mode doesn't support the suspend operation")
-
-	err = ValidateRayJobSpec(&rayv1.RayJob{
-		Spec: rayv1.RayJobSpec{
-			RuntimeEnvYAML: "invalid_yaml_str",
-			RayClusterSpec: createBasicRayClusterSpec(),
+		{
+			name: "the ClusterSelector mode doesn't support the suspend operation",
+			spec: rayv1.RayJobSpec{
+				Suspend:                  true,
+				ShutdownAfterJobFinishes: true,
+				ClusterSelector: map[string]string{
+					"key": "value",
+				},
+			},
+			expectError: true,
 		},
-	})
-	require.ErrorContains(t, err, "failed to unmarshal RuntimeEnvYAML")
-
-	err = ValidateRayJobSpec(&rayv1.RayJob{
-		Spec: rayv1.RayJobSpec{
-			BackoffLimit:   ptr.To[int32](-1),
-			RayClusterSpec: createBasicRayClusterSpec(),
+		{
+			name: "failed to unmarshal RuntimeEnvYAML",
+			spec: rayv1.RayJobSpec{
+				RuntimeEnvYAML: "invalid_yaml_str",
+				RayClusterSpec: createBasicRayClusterSpec(),
+			},
+			expectError: true,
 		},
-	})
-	require.ErrorContains(t, err, "backoffLimit must be a positive integer")
-
-	err = ValidateRayJobSpec(&rayv1.RayJob{
-		Spec: rayv1.RayJobSpec{
-			DeletionPolicy:           ptr.To(rayv1.DeleteClusterDeletionPolicy),
-			ShutdownAfterJobFinishes: true,
-			RayClusterSpec:           createBasicRayClusterSpec(),
+		{
+			name: "backoffLimit must be a positive integer",
+			spec: rayv1.RayJobSpec{
+				BackoffLimit:   ptr.To[int32](-1),
+				RayClusterSpec: createBasicRayClusterSpec(),
+			},
+			expectError: true,
 		},
-	})
-	require.ErrorContains(t, err, "RayJobDeletionPolicy feature gate must be enabled to use the DeletionPolicy feature")
-
-	features.SetFeatureGateDuringTest(t, features.RayJobDeletionPolicy, true)
-
-	err = ValidateRayJobSpec(&rayv1.RayJob{
-		Spec: rayv1.RayJobSpec{
-			DeletionPolicy:  ptr.To(rayv1.DeleteClusterDeletionPolicy),
-			ClusterSelector: map[string]string{"key": "value"},
+		{
+			name: "RayJobDeletionPolicy feature gate must be enabled to use the DeletionPolicy feature",
+			spec: rayv1.RayJobSpec{
+				DeletionPolicy:           ptr.To(rayv1.DeleteClusterDeletionPolicy),
+				ShutdownAfterJobFinishes: true,
+				RayClusterSpec:           createBasicRayClusterSpec(),
+			},
+			expectError: true,
 		},
-	})
-	require.ErrorContains(t, err, "the ClusterSelector mode doesn't support DeletionPolicy=DeleteCluster")
-
-	err = ValidateRayJobSpec(&rayv1.RayJob{
-		Spec: rayv1.RayJobSpec{
-			DeletionPolicy:  ptr.To(rayv1.DeleteWorkersDeletionPolicy),
-			ClusterSelector: map[string]string{"key": "value"},
+		{
+			name: "the ClusterSelector mode doesn't support DeletionPolicy=DeleteCluster",
+			beforeHook: func() {
+				// this feature gate is applied to the all following tests
+				features.SetFeatureGateDuringTest(t, features.RayJobDeletionPolicy, true)
+			},
```

Comment:
There is a before hook to apply the feature gate for the following tests.

Reply:
I think it's better to split this test into two tests. It's really bad if one test case has side effects that affect the following ones.

Reply:
In the splitted test, it should enable the feature gate, run some testcases, and then turn off the feature gate no matter whether are test failures or not.